WorkspaceServer.Tests/Kernel/CSharpKernelTestBase.cs

```diff
@@ -1,49 +1,21 @@
-// Copyright (c) .NET Foundation and contributors. All rights reserved.
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Collections.Generic;
 using Microsoft.DotNet.Interactive;
-using Microsoft.DotNet.Interactive.Events;
-using Pocket;
 using WorkspaceServer.Kernel;
 using Xunit.Abstractions;
-using System.Reactive.Linq;
-using System.Reactive;
 
 namespace WorkspaceServer.Tests.Kernel
 {
-    public class CSharpKernelTestBase : IDisposable
+    public abstract class CSharpKernelTestBase : KernelTestBase
     {
-        public CSharpKernelTestBase(ITestOutputHelper output)
+        public CSharpKernelTestBase(ITestOutputHelper output) : base(output)
         {
-            DisposeAfterTest(output.SubscribeToPocketLogger());
         }
 
-        protected CSharpKernel CreateKernel()
+        protected override KernelBase CreateBaseKernel()
         {
-            var kernel = new CSharpKernel()
-                         .UseDefaultRendering()
-                         .LogEventsToPocketLogger();
-
-            DisposeAfterTest(
-                kernel.KernelEvents.Timestamp().Subscribe(KernelEvents.Add));
-
-            return kernel;
-        }
-
-        private readonly CompositeDisposable _disposables = new CompositeDisposable();
-
-        protected IList<Timestamped<IKernelEvent>> KernelEvents { get; } = new List<Timestamped<IKernelEvent>>();
-
-        protected void DisposeAfterTest(IDisposable disposable)
-        {
-            _disposables.Add(disposable);
-        }
-
-        public void Dispose()
-        {
-            _disposables?.Dispose();
+            return new CSharpKernel();
```

Comment:
I have some changes in flight here as well and I'm leaning toward always using the same default set of extensions here as we use in the app at runtime. There's really only one supported configuration at the moment so we should test that.

Reply:
Currently I think the `.UseDefaultRendering()` extension should stay only in `CSharpKernelTestBase.cs` because the F# kernel doesn't support any of that yet since it's really just a thin wrapper/scraper around `fsi`.

The other alternative is to broaden the extension methods to switch on specific types like so:

``` csharp
public static KernelBase UseDefaultRendering(this KernelBase kernel)
{
    switch (kernel)
    {
        case CSharpKernel csharpKernel:
            csharpKernel.UseDefaultRendering();
            break;
        case FSharpKernel fsharpKernel:
            // noop
            break;
    }

    return kernel;
}

public static KernelBase UseDefaultRendering(this CSharpKernel kernel)
{
    // do C#-specific stuff
}
```

Thoughts?

Reply:
Rather than change the type that `UseDefaultRendering` accepts, maybe we should add an overload for `FSharpKernel`. The implication of the proposed method is that it should work for all kernel types, which isn't really possible since we can't know all of the possible implementations of `KernelBase`.

Reply:
I added `void SetDefaultRendering()` to `IKernel` so C# can implement it and F#'s is currently a noop.

Also updated to reflect your recent changes to `master`.

Reply:
I don't think that belongs on `IKernel`. Rendering is a fairly separate concern.